datafusion/physical-expr/src/window/window_frame_state.rs

```diff
@@ -85,6 +88,7 @@ impl<'a> WindowFrameContext<'a> {
                 sort_options,
                 length,
                 idx,
+                last_range,
```

Comment:
The last_range being added everywhere is a performance optimization to not throw away the index which was previously calculated?

Reply:
Yes, as it goes through batches, the search picks up from where it left off before. This works for fixed-boundary RANGE frames (which is what Datafusion supports).